src/main/java/org/sonarlint/intellij/SonarApplication.java

```diff
@@ -26,31 +26,47 @@
 import com.intellij.notification.NotificationGroup;
 import com.intellij.openapi.application.PathManager;
 import com.intellij.openapi.components.ApplicationComponent;
+import com.intellij.openapi.extensions.ExtensionPoint;
 import com.intellij.openapi.extensions.Extensions;
 import com.intellij.openapi.extensions.PluginId;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import org.jetbrains.annotations.NotNull;
 import org.sonarlint.intellij.core.SonarLintProjectNotifications;
 import org.sonarlint.intellij.core.SonarQubeEventNotifications;
 import org.sonarlint.intellij.editor.SonarExternalAnnotator;
 import org.sonarsource.sonarlint.core.client.api.util.FileUtils;
 
+import static org.sonarsource.sonarlint.shaded.com.google.common.base.Predicates.not;
+
 public class SonarApplication extends ApplicationComponent.Adapter {
   private IdeaPluginDescriptor plugin;
+  private ConcurrentMap<Language, LanguageExtensionPoint> annotatorsByLanguage;
```

Comment:
Not sure using Language instance as key is a good thing. The class doesn't implement hashCode. If you look at IJ code, they are either using the class, or the ID as key. If the same language is unloaded/reloaded, the instance may be different, even if this is the same language.

Reply:
In fact, maybe in this case we want to unregister/register again the annotator, for the new Language instance.